src/core/src/pass/serialize.cpp

```diff
@@ -1224,21 +1224,34 @@ void ngfunction_2_ir(pugi::xml_node& netXml,
     }
 }
 
+const std::filesystem::path check_path_safety(const std::filesystem::path& path) {
+    const bool contains_dotdot = std::any_of(path.begin(), path.end(), [](const auto& part) {
+        return part == "..";
+    });
+
+    OPENVINO_ASSERT(contains_dotdot == false,
+                    "Invalid file path: path contains parent directory reference '..': ",
+                    path);
+
+    OPENVINO_ASSERT(std::filesystem::is_symlink(path) == false, "Path must not refer to a symbolic link: ", path);
+    return path;
+}
+
 const std::filesystem::path valid_xml_path(const std::filesystem::path& path) {
     OPENVINO_ASSERT(path.extension() == ".xml",
-                    "Path for xml file doesn't contains file name with 'xml' extension: \"",
-                    path,
-                    "\"");
-    return path;
+                    "Path for xml file doesn't contains file name with 'xml' extension: ",
+                    path);
+
+    return check_path_safety(path);
 }
 
 std::filesystem::path provide_bin_path(const std::filesystem::path& xml_path, const std::filesystem::path& bin_path) {
     if (bin_path.empty()) {
         auto path = xml_path;
         path.replace_extension(".bin");
-        return path;
+        return check_path_safety(path);
     } else {
-        return bin_path;
+        return check_path_safety(bin_path);
```

Comment:
Check if not symlink and use sanitize path to remove traversal part.

Update doxy for class to mention that traversal part from path will be removed and symlinks are not allowed